### PR TITLE
Create Sphinx documentation for project

### PR DESCRIPTION
**Pull request by willingc**
Migrated content from markdown files to rst files. Created a sphinx documentation project that has a similar structure as other Jupyter docs.
cc/ @jdfreder @ellisonbg

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -0,0 +1,378 @@
...
+author = 'Jupyter Team'
+
+# The version info for the project you're documenting, acts as replacement for
+# |version| and |release|, also used in various other places throughout the
+# built documents.
+#
+# The short X.Y version.
+version = '4.1.1'
+# The full version, including alpha/beta/rc tags.
+release = '4.1.1'
```
Comment (jdfreder) [inline docs/source/conf.py]: We should get these from a centeralized location

**Comment by jdfreder** [comment]
Thanks @willingc !

**Comment by SylvainCorlay** [comment]
:+1:

File: docs/source/conf.py (hunk 1 above)
Comment (willingc) [inline docs/source/conf.py]: @jdfreder Yep. Good catch. I have some code that Min wrote that we've used in other docs to get version. I will find and open a separate PR.

**Comment by willingc** [comment]
Thanks @jdfreder and @SylvainCorlay

**Comment by willingc** [comment]
For the docs to fully deploy to ReadTheDocs, I will need sufficient privileges:
* on this repo for building docs on pushed changes to set hooks to update the build on RTD, and
* on RTD as a maintainer so I can update doc build settings.
Thanks.
Not sure who is the keeper of such powers :8ball: so I will ping @carreau @SylvainCorlay @jdfreder @minrk

**Comment by jdfreder** [comment]
@Carreau owns the RTD page, @Carreau can you add @willingc @SylvainCorlay and I?

@willingc I added you as a contributor to this repo.

**Comment by willingc** [comment]
Thanks @jdfreder :cake:

**Pull request merged**